src/app/bridge-btc/page.tsx

```diff
@@ -194,15 +194,20 @@ const BridgeBtc = () => {
 
   //const amount = new BN(10).mul(new BN(10).pow(new BN(12)));//.toString();
   const handleAmountChange = async (event: ChangeEvent<HTMLInputElement>) => {
-    const value = (event?.target as HTMLInputElement).value;
-    lg('handleAmountChange:', (event?.target as HTMLInputElement).value);
-    const out = Number.parseInt(value);
-    //TODO handle parseFloat instead
-    if (Number.isNaN(out)) {
-      console.error('parseInt failed');
-    } else {
-      setAmountIp(out);
+    let input = event?.target.value;
+    lg('handleAmountChange:', input);
+
+    const dpLen = count_decimals(input)
```

Comment:
can be extracted as amountFormatter(string): number | undefined

Reply:
`input` line needs to be there to be passed down further.
`console.log` or `lg` is for current debugging until production version.
`const dpLen` needs to be there to compare with MAX_DP then modify the `input` right there.
I have made a `checkNumInput()` to extract some code. okay?